pex/requirements.py

```diff
@@ -560,7 +488,11 @@ def _parse_requirement_line(
             parsed_url.fragment
         )
         project_name, extras, marker = (
-            project_name_extras_and_marker
+            (
+                project_name_extras_and_marker.project_name,
+                project_name_extras_and_marker.extras,
+                project_name_extras_and_marker.marker,
+            )
```

Comment:
```suggestion
            project_name_extras_and_marker.astuple()
```
Note that in this one we want `extras` to default to `None` instead of `()`, so we would have to modify `astuple` to allow overrides e.g. `project_name_extras_and_marker.astuple(extras=None)`. That's unless `extras` is correct as *either* `()` or `None` here. But `extras` differs from the other change in this file in that way.

Reply:
(see https://github.com/pantsbuild/pex/pull/1234#discussion_r577298669 -- i'm pretty confident this is a useful change, but please apply discretion)

Reply:
Yeah, good point. extras is in fact correct as either () or None since both the receiver at line 516 and the receiver at line 544 take `Optional[Iterable[str]]`. I had already been down this road as is shown by the fact I forgot to delete the currently unused typed astuple method here:
https://github.com/pantsbuild/pex/blob/bd542066f827676ede273a296d12b508a79c8ff0/pex/requirements.py#L343-L345

I'll just use that method, although the reason I didn't was I thought it didn't pull its weight and this repitition was more clear.

Reply:
Fixed.